src/foam/box/Runnable.js

```diff
@@ -0,0 +1,96 @@
+
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+foam.CLASS({
+  package: 'foam.box',
+  name: 'Runnable',
+
+  documentation: `An asynchronous computation that produces one or more outputs,
+      and sends them to its outputBox, reporting any errors to errorBox.`,
+
+  requires: [ 'foam.box.Message' ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'ioRelationshipType',
+      documentation: 'The n:m relationship type of input-to-output.',
+      value: '1:1'
```

Comment:
What uses this property? What if `n` or `m` is unknown? For example, what if I input one file which can result in a variable (`m`) number of computations?